app/code/Piimega/Maksuturva/Model/Gateway/Base.php

```diff
@@ -110,55 +108,4 @@ protected function getPostResponse($url, $data, $timeout = 120)
 
         return $res;
     }
-
-    protected function _verifyStatusQueryResponse($data)
-    {
-        $hashFields = array(
-            "pmtq_action",
-            "pmtq_version",
-            "pmtq_sellerid",
-            "pmtq_id",
-            "pmtq_amount",
-            "pmtq_returncode",
-            "pmtq_returntext",
-            "pmtq_sellercosts",
-            "pmtq_paymentmethod",
-            "pmtq_escrow",
-            "pmtq_certification",
-            "pmtq_paymentdate"
-        );
-
-        $optionalFields = array(
-            "pmtq_sellercosts",
-            "pmtq_paymentmethod",
-            "pmtq_escrow",
-            "pmtq_certification",
-            "pmtq_paymentdate"
-        );
-
-        $hashString = "";
-        foreach ($hashFields as $hashField) {
```

Comment:
Actually this part may have to be returned and just the hash calculation part be removed...